src/main/java/ch/njol/skript/expressions/ExprLoopIteration.java

```diff
@@ -0,0 +1,119 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.LoopSection;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.TriggerSection;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Loop Iteration")
+@Description("Returns the loop's current iteration count (for both normal and while loops).")
+@Examples({
+	"while player is online:",
+		"\tgive player 1 stone",
+		"\twait 5 ticks",
+		"\tif loop-counter > 30:",
+			"\t\tstop loop",
+	"",
+	"loop {top-balances::*}:",
+		"\tif loop-iteration <= 10:",
+		"\t\tbroadcast \"##%loop-iteration% %loop-index% has $%loop-value%\"",
+})
+@Since("INSERT VERSION")
+public class ExprLoopIteration extends SimpleExpression<Long> {
+
+	static {
+		Skript.registerExpression(ExprLoopIteration.class, Long.class, ExpressionType.SIMPLE, "[the] loop(-| )(counter|iteration)[-%-*number%]");
```

Comment:
This might be better as ExpressionType.COMBINED (although the expression is optional)
@TPGamesNL thoughts?

Reply:
I think this should stay SIMPLE, because the expression in it, `%-*number%`, only allows literals and is therefore not expensive to parse